internal_transfers/actions/src/accounting.ts

```diff
@@ -62,24 +62,46 @@ export async function simulateSolverSolution(
     console.log(`batch ${transaction.hash} was not internalized.`);
     return null;
   }
-
-  const { full, reduced } = await simulateBoth(simulator, {
-    full: competition.fullCallData,
-    reduced: competition.reducedCallData,
-    common: {
-      contractAddress: SETTLEMENT_CONTRACT_ADDRESS,
-      sender: solverAddress,
-      value: "0",
-    },
-    startBlock: competition.simulationBlock,
-  });
-
-  return {
-    txHash: transaction.hash,
-    winningSettlement: competition,
-    full,
-    reduced,
-  };
+  // Try all blocks between claimed simulation block and mined block.
+  const numAttempts = transaction.blockNumber - competition.simulationBlock + 1;
+  try {
+    const { full, reduced } = await simulateBoth(
+      simulator,
+      {
+        full: competition.fullCallData,
+        reduced: competition.reducedCallData,
+        common: {
+          contractAddress: SETTLEMENT_CONTRACT_ADDRESS,
+          sender: solverAddress,
+          value: "0",
+        },
+        startBlock: competition.simulationBlock,
+      },
+      numAttempts
+    );
+    return {
+      txHash: transaction.hash,
+      winningSettlement: competition,
+      full,
+      reduced,
+    };
+  } catch (error: any) {
+    // If we fail this many simulations, we will just have to assume there is none.
+    console.error(error.errorMessage);
+    const failedSimulation = {
+      simulationID: `failed all ${numAttempts} simulation attempts`,
+      blockNumber: -1,
+      gasUsed: -1,
```

Comment:
I'd be willing to consider using Null in these two places (but it would result in a slightly bigger change).

Reply:
The nice thing is that we can identify them in this way (and so far - after five days there is still only one) m. However they also result in empty logs which also means we assumed there was no token imbalance and this becomes a permanent assumption after solver payout has be done for the week. 

We absolutely must use some indication that these have failed and undergone a sufficient number of attempts. We can always accumulate these and come back to them (as long as they are identifiable). Whether it be null or -1 is a different question.

Reply:
```suggestion
      blockNumber: -1,
      gasUsed: 0,
```